pyproject.toml

```diff
@@ -51,6 +51,49 @@ write_to = "astropy/_version.py"
     group_by_package = true
     indented_import_headings = false
 
+[tool.coverage]
+
+    [tool.coverage.run]
+        omit = [
+            "astropy/__init__*",
+            "astropy/conftest.py",
+            "astropy/*setup*",
+            "astropy/*/tests/*",
+            "astropy/tests/test_*",
+            "astropy/extern/*",
+            "astropy/utils/compat/*",
+            "astropy/version*",
+            "astropy/wcs/docstrings*",
+            "astropy/_erfa/*",
+            "*/astropy/__init__*",
+            "*/astropy/conftest.py",
+            "*/astropy/*setup*",
+            "*/astropy/*/tests/*",
+            "*/astropy/tests/test_*",
+            "*/astropy/extern/*",
+            "*/astropy/utils/compat/*",
+            "*/astropy/version*",
+            "*/astropy/wcs/docstrings*",
+            "*/astropy/_erfa/*",
+        ]
+
+    [tool.coverage.report]
+        exclude_lines = [
+            # Have to re-enable the standard pragma
+            "pragma: no cover",
+            # Don't complain about packages we have installed
+            "except ImportError",
+            # Don't complain if tests don't hit defensive assertion code:
+            "raise AssertionError",
+            "raise NotImplementedError",
+            # Don't complain about script hooks
+            "'def main(.*):'",
```

Comment:
My TOML-fu is weak. Do we need raw strings for pattern matching to work properly or not? In setup.cfg, it is defined as plain text, so we don't have this issue.

Reply:
A good Q. My toml-fu is also pretty weak. In A-B testing coverage doesn't seem to drop, so I guess that's an answer?